### PR TITLE
feat: Java client supports creating roles

### LINKED ISSUE #25297: Java client supports role creation

### PR DESCRIPTION
**Pull request by matthewBearCamunda**
## Description

This PR introduces support for creating roles in the Java client.

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #25297

### CODE REVIEW

File: clients/java/src/main/java/io/camunda/zeebe/client/api/command/CreateRoleCommandStep1.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import io.camunda.zeebe.client.api.response.CreateRoleResponse;
+
+public interface CreateRoleCommandStep1 extends FinalCommandStep<CreateRoleResponse> {
+  /**
+   * Set the role ID for the role to be created.
+   *
+   * @param roleId the unique role ID
+   * @return the builder for this command
+   */
+  CreateRoleCommandStep1 roleId(String roleId);
```
Comment (Ben-Sheppard) [inline clients/java/src/main/java/io/camunda/zeebe/client/api/command/CreateRoleCommandStep1.java:27]: 🔧 I think this can be removed as its not used underlying code

File: clients/java/src/main/java/io/camunda/zeebe/client/api/command/CreateRoleCommandStep1.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.zeebe.client.api.command;
+
+import io.camunda.zeebe.client.api.response.CreateRoleResponse;
+
+public interface CreateRoleCommandStep1 extends FinalCommandStep<CreateRoleResponse> {
+  /**
+   * Set the name to create user with.
```
Comment (nicpuppa) [inline clients/java/src/main/java/io/camunda/zeebe/client/api/command/CreateRoleCommandStep1.java:22]:
```suggestion
   * Set the name to create role with.
```

File: clients/java/src/main/java/io/camunda/zeebe/client/impl/command/CreateRoleCommandImpl.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+  @Override
+  public FinalCommandStep<CreateRoleResponse> requestTimeout(final Duration requestTimeout) {
+    httpRequestConfig.setResponseTimeout(requestTimeout.toMillis(), TimeUnit.MILLISECONDS);
+    return this;
+  }
+
+  @Override
+  public ZeebeFuture<CreateRoleResponse> send() {
+    ArgumentUtil.ensureNotNull("name", request.getName());
```
Comment (nicpuppa) [inline clients/java/src/main/java/io/camunda/zeebe/client/impl/command/CreateRoleCommandImpl.java:60]: 🔧 This can be removed, as you are already check the name in the method above

File: clients/java/src/test/java/io/camunda/zeebe/client/role/CreateRoleTest.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import io.camunda.zeebe.client.api.command.ProblemException;
+import io.camunda.zeebe.client.protocol.rest.ProblemDetail;
+import io.camunda.zeebe.client.protocol.rest.RoleCreateRequest;
+import io.camunda.zeebe.client.util.ClientRestTest;
+import org.junit.jupiter.api.Test;
+
+public class CreateRoleTest extends ClientRestTest {
```
Comment (nicpuppa) [inline clients/java/src/test/java/io/camunda/zeebe/client/role/CreateRoleTest.java:28]: ❌ I'm missing the test case where an exception is raised for null name

**Review by nicpuppa — CHANGES_REQUESTED** [review]
❌ You should add an AuthorizationIT test to verify the requests gets rejected if the user unauthorized? You can use the UserCreateAuthorizationIT as inspiration.

File: clients/java/src/main/java/io/camunda/zeebe/client/impl/command/CreateRoleCommandImpl.java (hunk 3 above)
Comment (Ben-Sheppard) [reply clients/java/src/main/java/io/camunda/zeebe/client/impl/command/CreateRoleCommandImpl.java:60]: We seem to also use this pattern in other commands i.e. `CreateUserCommandStep1`

**Review by Ben-Sheppard — APPROVED** [review]
Looks good to me though as @nicpuppa requested changes it would be good to (maybe even required) get his thoughts :)

File: clients/java/src/main/java/io/camunda/zeebe/client/impl/command/CreateRoleCommandImpl.java (hunk 3 above)
Comment (nicpuppa) [reply clients/java/src/main/java/io/camunda/zeebe/client/impl/command/CreateRoleCommandImpl.java:60]: Yes because the `send()` step is on the same level of the other steps. My bad.

**Review by nicpuppa — APPROVED** [review]
